src/foam/nanos/auth/PasswordPolicy.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * @license
+ * Copyright 2019 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.auth',
+  name: 'PasswordPolicy',
+
+  implements: [
+    'foam.nanos.auth.EnabledAware'
+  ],
+
+  documentation: 'A Password Policy for a Group.',
+
+  constants: [
+    {
+      name: 'MIN_PASSWORD_LENGTH',
+      type: 'Long',
+      value: 6
+    }
+  ],
+
+  properties: [
+    {
+      class: 'Boolean',
+      name: 'enabled'
+    },
+    {
+      class: 'Long',
+      name: 'minLength',
+      documentation: 'Minimum length of the password.',
+      min: 6,
+      factory: function() {
+        return MIN_PASSWORD_LENGTH;
+      },
+      javaFactory: `
+        return MIN_PASSWORD_LENGTH;
+      `
+    },
+    {
+      class: 'Long',
+      name: 'priorPasswordsToCheckCount',
+      min: 0,
+      max: 10,
+      documentation: `
+        User passwords are hashed and the hash is kept for historical record up to a maximum of 10 previous entries per user. 
+        Setting priorPasswordsToCheckCount to a value greater than zero will enable a check on that many previous password entries
+        when resetting the users passwords to prevent them from using the same password again.
+        `
+    },
+  ],
+
+  javaImports: [
+    'foam.nanos.auth.User',
+    'foam.util.Password',
+    'foam.util.SafetyUtil'
+  ],
+
+  methods: [
+    {
+      name: 'validate',
+      async: true,
```

Comment:
Why is this async?

Reply:
I think it was a copy paste error. I'll remove it.